reconcile/alertmanager_config_validator.py

```diff
@@ -0,0 +1,88 @@
+import sys
+import semver
+import base64
+import logging
+
+import reconcile.utils.gql as gql
+import reconcile.utils.threaded as threaded
+import reconcile.utils.amtool as amtool
+import reconcile.openshift_resources_base as orb
+
+from reconcile.status import ExitCodes
+
+QONTRACT_INTEGRATION = 'alertmanager_config_validator'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+ALERTMANAGER_SECRET_PATH = '/observability/alertmanager/' \
+                           'alertmanager-instance.secret.yaml'
+ALERTMANAGER_SECRET_KEY = 'alertmanager.yaml'
+
+
+def check_config(config):
+    '''Helper function to run amtool check-config via threaded.run'''
+    config['check_result'] = amtool.check_config(config['data'])
+    return config
+
+
+def get_config_data(namespace_info):
+    '''Returns a dict with the alertmanager config from the cluster refered
+       in the namespace info'''
+
+    openshift_resources = namespace_info.get('openshiftResources')
+
+    if not openshift_resources:
+        return
+
+    for r in openshift_resources:
+        path = r['path']
+        if path != ALERTMANAGER_SECRET_PATH:
+            continue
+
+        openshift_resource = \
+            orb.fetch_openshift_resource(resource=r, parent=namespace_info)
+
+        try:
+            encoded = openshift_resource.body['data'][ALERTMANAGER_SECRET_KEY]
+            data = base64.b64decode(encoded).decode('utf-8')
+        except KeyError:
+            logging.warning('No data found in config secret for key '
+                            f'{ALERTMANAGER_SECRET_KEY}')
+            return
+
+        return {'cluster': namespace_info['cluster']['name'], 'data': data}
+
+
+def run(dry_run, thread_pool_size=20, cluster_name=None):
+    orb.QONTRACT_INTEGRATION = QONTRACT_INTEGRATION
+    orb.QONTRACT_INTEGRATION_VERSION = QONTRACT_INTEGRATION_VERSION
+
+    gqlapi = gql.get_api()
+
+    configs = []
+    for namespace_info in gqlapi.query(orb.NAMESPACES_QUERY)['namespaces']:
```

Comment:
this is the 3rd duplicate of `gqlapi.query(orb.NAMESPACES_QUERY)['namespaces']`. is it time to extract this to a `get_namespaces` function in `orb`? just a nit

Reply:
Good point. I'll do it

Reply:
maybe even only iterate over namespaces with are `observabilityNamespace`?

Reply:
Hmmm, good idea. I didn't remember about that property. I'll take a look into that. That removes another hardcoded value which is always a good idea